<launch>


    <!-- global parameters -->
    <param name="latitude_reference" value="32.5939"/>
    <param name="longitude_reference" value="-85.4867"/>

    <!-- IMU -->
    <node pkg="xbow440" type="xbow440_node" name="xbow440_node" output="screen" >
        <param name="port" value="/dev/ttyS2" />
        <param name="baudrate" value="57600" />
    </node>

    <!-- Novatel -->
    <include file="$(find path_duplication_launch)/launch/novatel_front.launch"/>
    <include file="$(find path_duplication_launch)/launch/novatel_rear.launch"/>

    <!-- DRTK -->
    <include file="$(find path_duplication_launch)/launch/drtk.launch"/>
    <include file="$(find path_duplication_launch)/launch/drtk.launch"/>

    <!-- TDCP -->
    <node pkg="tdcp" type="tdcp_node" name="tdcp_node" output="screen" >
        <param name="ephemeris_topic" value="/novatel_rear/gpsEphemerisTagged" type="string"/>
        <param name="raw_measurements_topic" value="/novatel_rear/rawMeasurementsTagged" type="string"/>
    </node>

    <!-- GPS/INS -->
    <node pkg="ins_ekf" type="gps_ins_node" name="gps_ins_node" output="screen" >
        <rosparam file="$(find path_duplication_launch)/config/gps_ins_crossbow_parameters.yaml" command="load" />

        <param name="gps_topic_name" value="/novatel_rear/odom" type="string"/>
        <param name="imu_topic_name" value="/xbow440_node/imu/data" type="string"/>

    </node>

    <!-- Path Duplication -->
    <node pkg="path_duplication" type="path_duplication_node" name="path_duplication_node" output="screen" required="true">
        <param name="drtk_topic" value="/drtk_node/drtkOutput" type="string"/>
        <param name="tdcp_topic" value="/tdcp_node/tdcpOutput" type="string"/>
        <param name="queue_size" value="0"/>
    </node>

    <!-- Prowler Steer Control -->
    <node pkg="prowler_steer_control" type="prowler_steer_control_node.py" name="prowler_steer_control_node" output="screen" required="true">
        <param name="control_reference_topic" value="/path_duplication_node/headingControlReference" type="string"/>
        <param name="velocity_topic" value="/novatel_rear/odom" type="string"/>
    </node>

</launch>
